tidb-cloud: add doc about using aws dms migrate data from oracle to tidb (#11501)
pingcap/docs: tidb-cloud/migrate-from-oracle-using-aws-dms.md

- [TiDB Developer Guide](https://docs.pingcap.com/tidbcloud/dev-guide-overview)
- [AWS DMS Documentation](https://docs.aws.amazon.com/dms/latest/userguide/CHAP_GettingStarted.html)

## What is AWS DMS?

AWS Database Migration Service (AWS DMS) is a cloud service that makes it possible to migrate relational databases, data warehouses, NoSQL databases, and other types of data stores.

## Why use AWS DMS?

If you want to migrate data from from heterogeneous databases, such as PostgreSQL, Oracle, and SQL Server to TiDB Cloud, it is recommended to use AWS Database Migration Service (AWS DMS).

## Deployment architecture

At a high level, follow the following steps:

- Setting up source AWS Oracle RDS.
- Setting up target [TiDB Cloud (Serverless) Tier](https://tidbcloud.com/console/clusters/create-cluster).
- Setting up data migration (full load) using AWS DMS.

The following diagram illustrates the high-level architecture.

![Architecture](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-0.png)

## Prerequisites

You might need the following prerequisites:

- [AWS Cloud Account](https://aws.amazon.com)
- [TiDB Cloud Account](https://tidbcloud.com)
- [DBeaver](https://dbeaver.io/)

> Next, you will learn how to use DMS tools to migrate AWS Oracle RDS data to TiDB.

Reviewer: ```suggestion
Next, you will learn how to use DMS tools to migrateAmazon RDS for Oracle data to TiDB.
```

## Step 1. Create a VPC

> Create an AWS VPC. We will create Oracle RDS and DMS instances in this VPC later.

Reviewer: ```suggestion
Create an AWS VPC. You will need to create Oracle RDS and DMS instances in this VPC later.
```

![Create VPC](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-1.png)

## Step 2. Create Oracle RDS

> Create an AWS Oracle RDS in VPC, and remember the password and give it public access. You must enable public access to use the AWS Schema Conversion Tool. Attention, giving public access in product environment is not a best practice.

Reviewer: ```suggestion
Create an AWS Oracle RDS in VPC, and remember the password and give it public access. You must enable public access to use the AWS Schema Conversion Tool. Note that giving public access in the product environment is not a best practice.
```

![Create Oracle RDS](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-2.png)

> ## Step 3. Prepare Table Data In Oracle

Reviewer: ```suggestion
## Step 3. Prepare table data in Oracle
```

> In this step, we will insert some data into Oracle RDS. We will use the github event dataset, and you can download the data from [GH Archive](https://gharchive.org/). We already download 10000 rows data and you can use following SQL script to execute in Oracle.

Reviewer: ```suggestion
In this step, insert some data into Oracle RDS. Use the github event dataset, and you can download the data from [GH Archive](https://gharchive.org/). It contains 10000 rows of data and you can use following SQL script to execute in Oracle.
```

---
title: Using AWS DMS To Migrate Data From AWS RDS To TiDB Cloud (Serverless) Tier
---

# Using AWS DMS To Migrate Data From AWS RDS To TiDB Cloud (Serverless) Tier

This document describes a step-by-step example of how to migrate data from AWS Oracle RDS to [TiDB Cloud Serverless Tier](https://tidbcloud.com/console/clusters/create-cluster) using AWS Database Migration Service (AWS DMS).

If you are interested in learning more about TiDB and AWS DMS, you can find some useful links as follows:

- [TiDB Cloud](https://docs.pingcap.com/tidbcloud/)
- [TiDB Developer Guide](https://docs.pingcap.com/tidbcloud/dev-guide-overview)
- [AWS DMS Documentation](https://docs.aws.amazon.com/dms/latest/userguide/CHAP_GettingStarted.html)

## What is AWS DMS?

AWS Database Migration Service (AWS DMS) is a cloud service that makes it possible to migrate relational databases, data warehouses, NoSQL databases, and other types of data stores.

## Why use AWS DMS?

If you want to migrate heterogeneous databases, such as PostgreSQL, Oracle, and SQL Server to TiDB Cloud, it is recommended to use AWS Database Migration Service (AWS DMS).

## Deployment architecture

> At a high level, this post includes the following steps:

Author: done.

- Setting up source AWS Oracle RDS.
- Setting up target [TiDB Cloud (Serverless) Tier](https://tidbcloud.com/console/clusters/create-cluster).
- Setting up data migration (full load) using AWS DMS.

The following diagram illustrates the high-level architecture.

> ![Architecture](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-0.png)

Author: I already add a tidb logo, but I don't know/have the  Amazon RDS logo.

## Prerequisites

You might need the following prerequisites:

- [AWS Cloud Account](https://aws.amazon.com)
- [TiDB Cloud Account](https://tidbcloud.com)
- [DBeaver](https://dbeaver.io/)

Next, you will learn how to use DMS tools to migrate AWS Oracle RDS data to TiDB.

## Step 1. Create a VPC

Create an AWS VPC. We will create Oracle RDS and DMS instances in this VPC later.

![Create VPC](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-1.png)

## Step 2. Create Oracle RDS

> Create an AWS Oracle RDS in VPC, and remember the password and give it public access. You must enable public access to use the AWS Schema Conversion Tool.

Author: done

## Step 8. Create DMS Database Migration Task

On AWS Management Console, Create DMS Migration Task and specify the **Selection rules**:

![Create AWS DMS migration task](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-11.png)

![AWS DMS migration task selection rules](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-12.png)

Then create the task, start it, and then wait for the task to finish.

Then click the **Table statistics** to see the synced table, remember the schema name is ADMIN:

![Check AWS DMS migration task](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-13.png)

## Step 9. Check data In Downstream TiDB Cloud Serverless Tier

Connect to [TiDB Cloud Serverless Tier](https://tidbcloud.com/console/clusters/create-cluster) and check the `admin.github_event` table data, as you can see, DMS successfully synced table github_events and 10000 rows data.

![Check Data In TiDB](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-14.png)

## Conclusion

> With AWS DMS, you can successfully migrate data from upstream AWS RDS(MySQL, Oracle) to [TiDB Cloud Serverless Tier](https://tidbcloud.com/console/clusters/create-cluster) With the following exceptions:

Reviewer 2: Probably change  - you can successfully migrate data from upstream AWS RDS(MySQL, Oracle) 
**to**
you can successfully migrate data from ANY upstream AWS RDS database in this case we used Oracle

Author: done

- when you migrate your schema, some charset/collation is not supported in TiDB, such as `utf8mb4_0900_ai_ci`. A complete list of [TiDB supported charset/collation doc can be found here](https://docs.pingcap.com/tidb/v6.3/character-set-and-collation).

> Here are some speed test result. In the following test, AWS Oracle RDS and AWS MySQL RDS are deployed in us-west-2a region, and both within the same VPC. [TiDB Cloud Serverless Tier](https://tidbcloud.com/console/clusters/create-cluster) is in us-west-2 region.

Reviewer 2: Here are some speed test result.

I think we should call it

Here are some performance benchmarking results with incremental data size.

Also remove AWS MySQL RDS

Author: I just remove it since I don't have more benchmark results...

| Source         | Target                     | Data Size | Time Cost |
| -------------- | -------------------------- | --------- | --------- |

> | AWS Oracle RDS | TiDB Cloud Serverless Tier | 0.1GB     | 14s       |

Reviewer 2: Since we only showed Oracle RDS can we remove MySQL, and can we show more benchmarking with increasing Data Size ?

Author: I just remove it since I don't have more benchmark results...

    ![Create task](/media/tidb-cloud/aws-dms-tidb-cloud/aws-dms-to-tidb-cloud-create-task.png)

2. Create a database migration task and specify the **Selection rules**:

    ![Create AWS DMS migration task](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-11.png)

    ![AWS DMS migration task selection rules](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-12.png)

3. Create the task, start it, and then wait for the task to finish.

4. Click the **Table statistics** to see the synced table. The schema name is ADMIN.

    ![Check AWS DMS migration task](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-13.png)

## Step 9. Check data in the downstream TiDB cluster

Connect to [TiDB Cloud Serverless Tier](https://tidbcloud.com/console/clusters/create-cluster) and check the `admin.github_event` table data. As shown in the following screenshot, DMS successfully migrated table `github_events` and 10000 rows of data.

![Check Data In TiDB](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-14.png)

> ## Limitation

Author: Yes, Since cloud PM  said that the specifications are different, the data samples are different, and the migration performance will be very different, so it is not recommended to put it in doc.

- [TiDB Cloud](https://docs.pingcap.com/tidbcloud/)
- [TiDB Developer Guide](https://docs.pingcap.com/tidbcloud/dev-guide-overview)
- [AWS DMS Documentation](https://docs.aws.amazon.com/dms/latest/userguide/CHAP_GettingStarted.html)

## Why use AWS DMS?

AWS Database Migration Service (AWS DMS) is a cloud service that makes it possible to migrate relational databases, data warehouses, NoSQL databases, and other types of data stores.

If you want to migrate data from from heterogeneous databases, such as PostgreSQL, Oracle, and SQL Server to TiDB Cloud, it is recommended to use AWS Database Migration Service (AWS DMS).

## Deployment architecture

At a high level, follow the following steps:

1. Set up source Amazon RDS for Oracle.
2. Set up the target [TiDB Cloud (Serverless) Tier](https://tidbcloud.com/console/clusters/create-cluster).
3. Set up data migration (full load) using AWS DMS.

The following diagram illustrates the high-level architecture.

![Architecture](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-0.png)

## Prerequisites

You might need the following prerequisites:

> - [AWS DMS Prerequisites](/tidb-cloud/migrate-from-mysql-using-aws-dms.md#Prerequisites)

Author: link to here: https://github.com/pingcap/docs/pull/11438/files#diff-929b11aba15553b7b3f1dc934fc150cf892cf8d394c7ff77104e89c8bad95b45R18

Next, you will learn how to use DMS tools to migrate data from Amazon RDS for Oracle into TiDB Cloud.

## Step 1. Create a VPC

Log in to the [AWS console](https://console.aws.amazon.com/vpc/home#vpcs:) and create an AWS VPC. You will need to create Oracle RDS and DMS instances in this VPC later.

For instructions about how to create a VPC, see [Creating a VPC](https://docs.aws.amazon.com/vpc/latest/userguide/working-with-vpcs.html#Create-VPC).

![Create VPC](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-1.png)

## Step 2. Create an Oracle DB instance

Create an Oracle DB instance in VPC, and remember the password and give it public access. You must enable public access to use the AWS Schema Conversion Tool. Note that giving public access in the product environment is not a best practice.

For instructions about how to create an Oracle DB instance, see [Creating an Oracle DB instance and connecting to a database on an Oracle DB instance](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_GettingStarted.CreatingConnecting.Oracle.html).

![Create Oracle RDS](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-2.png)

## Step 3. Prepare the table data in Oracle

> In this step, insert some data into Oracle DB instance. Use the github event dataset, and you can download the data from [GH Archive](https://gharchive.org/). It contains 10000 rows of data. You can use following SQL script to execute in Oracle.

Author: great, done.

## Step 4. Create a TiDB Cloud Serverless Tier cluster

1. Log in to the [TiDB Cloud console](https://tidbcloud.com/console/clusters). Navigate to the **Clusters** page for your project.

2. Create a free serverless TiDB tier.

    ![Create TiDB Cloud Serverless Tier](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-4.png)

3. After the cluster is created, click **Security Settings** to set the user password.

    ![Set TiDB Cloud Serverless Tier Password](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-5.png)

    ![Set TiDB Cloud Serverless Tier Password](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-6.png)

4. Click the **Generate** button and copy to generate the password. Note down the generated password. Then click the **Submit** button.

5. Click the **Connect** button to connect the serverless TiDB:

    ![Connect to TiDB Cloud Serverless Tier](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-7.png)

## Step 5. Create an AWS DMS replication instance

1. Go to the [Replication instances](https://console.aws.amazon.com/dms/v2/home#replicationInstances) page in the AWS DMS console, and switch to the corresponding region.

> 2. Create AWS DMS replication instance with `dms.t3.large` in VPC.

Author: No, This is the configuration I used for my demo. [DMS official doc](https://docs.aws.amazon.com/dms/latest/userguide/CHAP_GettingStarted.Replication.html) demo use `dms.t3.medium`.

3. After the cluster is created, click **Security Settings** to set the user password.

    ![Set TiDB Cloud Serverless Tier Password](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-5.png)

    ![Set TiDB Cloud Serverless Tier Password](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-6.png)

4. Click the **Generate** button generate the password. Note down the generated password. Click the **Submit** button.

5. Click the **Connect** button to connect your TiDB cluster:

    ![Connect to TiDB Cloud Serverless Tier](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-7.png)

## Step 5. Create an AWS DMS replication instance

1. Go to the [Replication instances](https://console.aws.amazon.com/dms/v2/home#replicationInstances) page in the AWS DMS console, and switch to the corresponding region.

2. Create an AWS DMS replication instance with `dms.t3.large` in the VPC.

    ![Create AWS DMS Instance](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-8.png)

## Step 6. Create DMS endpoints

1. In the [AWS DMS console](https://console.aws.amazon.com/dms/v2/home), click the replication instance that you just created.

> 2. Create the Oracle source endpoint and the TiDB target endpoint.

Reviewer 2: Creating oracle source endpoint and target TiDB is a bit confusing for me and here is why

     -  In step 1 you want the user to go to the DMS Console and Click the replication instance. Is this necessary I think the menu item on the left has the  navigation item that allows you to create end point isnt it ? I will send you a screen shot on Lark.

Author: This is unnecessary, I'll change it.

## Step 8. Create a database migration task

1. In the AWS DMS console, go to the [Data migration tasks](https://console.aws.amazon.com/dms/v2/home#tasks) page. Switch to your region. Then click **Create task** in the upper right corner of the window.

    ![Create task](/media/tidb-cloud/aws-dms-to-tidb-cloud-create-task.png)

2. Create a database migration task and specify the **Selection rules**:

    ![Create AWS DMS migration task](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-11.png)

    ![AWS DMS migration task selection rules](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-12.png)

3. Create the task, start it, and then wait for the task to finish.

4. Click the **Table statistics** to check the table. The schema name is `ADMIN`.

    ![Check AWS DMS migration task](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-13.png)

## Step 9. Check data in the downstream TiDB cluster

Connect to the [Serverless Tier cluster](https://tidbcloud.com/console/clusters/create-cluster) and check the `admin.github_event` table data. As shown in the following screenshot, DMS successfully migrated table `github_events` and 10000 rows of data.

> ![Check Data In TiDB](/media/tidb-cloud/aws-dms-from-oracle-to-tidb-14.png)

Reviewer 2: /media/tidb-cloud/aws-dms-from-oracle-to-tidb-14.png image has mysql ERROR 2013 please remove that from the screenshot .

Author: nice catch, done.
